scg-system/src/main/java/io/telicent/labels/servlets/LabelsQueryServlet.java

```
@@ -27,37 +37,120 @@ public LabelsQueryServlet(LabelsQueryService queryService) {
-            response.setCharacterEncoding(WebContent.charsetUTF8);
-            out.print(jsonOutput);
-        } catch (IOException ex) {
-            response.setStatus(HttpServletResponse.SC_UNPROCESSABLE_CONTENT);
+    ArrayNode populateNodeWithResults(List<TripleLabels> tripleLabels) {
+        ArrayNode resultNode = MAPPER.createArrayNode();
+        tripleLabels.forEach(tripleLabel -> {
+            resultNode.add(tripleLabel.toJSONNode());
+        });
+        return resultNode;
+    }
+
+    public static boolean isWildcardTriple(Triple triple) {
+        if (Node.ANY.equals(triple.getSubject())) {
+            return true;
+        } else if (Node.ANY.equals(triple.getPredicate())) {
+            return true;
+        } else return Node.ANY.equals(triple.getObject());
+    }
+
+    List<Triple> obtainTripleQueries(HttpServletRequest request) {
+        List<Triple> tripleList = new ArrayList<>();
+        try (final InputStream inputStream = request.getInputStream()) {
+                JsonNode rootNode = MAPPER.readTree(inputStream); // Read into a JsonNode
+                if (rootNode.isArray()) {
+                    for (JsonNode node : rootNode) {
+                        LabelsQuery query = MAPPER.convertValue(node, LabelsQuery.class); // Convert each node
+                        tripleList.add(getTriple(query));
+                    }
+                } else if (rootNode.isObject()) {
+                    LabelsQuery query = MAPPER.convertValue(rootNode, LabelsQuery.class); // Convert the root node
+                    tripleList.add(getTriple(query));
+                } else {
+                    // IGNORE?
+                    System.err.println("Invalid JSON format: Must be an object or an array.");
+                }
+
+            } catch (IOException exception) {
+                // Handle I/O errors
+                exception.printStackTrace(); // Replace with proper logging
```

Would it be better to throw the exception, catch in the `doPost` method and then return an error response? At least then the user would know something went wrong rather than their being no result.

---

It's for debugging so I'm in two minds.  I never like 500 errors but similarly hate not being clear as to what the issue is.

I'll have a play about and add some tests for that too.